demo/qat-autoreset.sh

```diff
@@ -1,12 +1,20 @@
 #!/usr/bin/env bash
 NODE_NAME="${NODE_NAME:-}"
 ENABLED_QAT_PF_PCIIDS=${ENABLED_QAT_PF_PCIIDS:-37c8 4940 4942 4944 4946}
-DEVS=$(for pf in $ENABLED_QAT_PF_PCIIDS; do lspci -n | grep -e "$pf" | grep -o -e "^\\S*"; done)
 
 AUTORESET_ENABLED="NONE"
 AUTORESET_ENABLED_FOUND="FALSE"
 AUTORESET_OPTIONS_LIST="on off"
 
+DEVS=""
+for DEV in $(realpath /sys/bus/pci/devices/*); do
+  for PF in $ENABLED_QAT_PF_PCIIDS; do
+    if grep -q "$PF" "$DEV"/device; then
+      DEVS="$DEVS $DEV"
+    fi
+  done
+done
```

Comment:
Optimized alternative doing just single `grep` invocation, for both of the scripts:
```suggestion
DEV_PATTERNS=$(for i in $ENABLED_QAT_PF_PCIIDS; do echo "-e $i"; done)
DEVS=$(dirname $(grep -l $DEV_PATTERNS /sys/bus/pci/devices/*/device))
```

(When `echo` is invoked without path, it's Bash internal, not an external command.)

Reply:
I wonder if those flags work with the toybox grep.

Reply:
Oh, and it could use also `-F` option to indicate given values are literal match strings, not regexps.

Reply:
> ```
> grep: /sys/bus/pci/devices/*/device: No such file or directory
> ```

Does  `realpath` work for that file pattern?

Reply:
> -l and -e are standard ones, not GNU extensions. If it does not implement them, it's broken.

Toybox versions are very limited in terms of functionality. The qat-initcontainer also doesn't contain bash, but a toybox shell.

> Does realpath work for that file pattern?

```
# realpath /sys/bus/pci/devices/*/device
realpath: /sys/bus/pci/devices/*/device: No such file or directory
```

Reply:
I explored the `DEV_PATTERNS` idea too but interestingly that results in:
```
# echo $DEV_PATTERNS
37c8 -e 4940 -e 4942 -e 4944 -e 4946
```

Reply:
`realpath: /sys/bus/pci/devices/*/device: No such file or directory`

@tkatila Wouldn't then what the original code does (`realpath /sys/bus/pci/devices/*`), also fail?

Reply:
> I explored the `DEV_PATTERNS` idea too but interestingly that results in:
> 
> ```
> # echo $DEV_PATTERNS
> 37c8 -e 4940 -e 4942 -e 4944 -e 4946
> ```

If it's indeed toybox shell instead of e.g. Bash, it's probably using separate Toybox `echo` command.  However, it's  pretty broken if it's *randomly* interpreting `"-e ..."` as `"-e" "..."` (`-e` option for `echo` enables backslash escapes). 

=> It may work better with extra space at start of the quoted string: `echo " -e $i"`

Btw. it could probably be `echo " -e 0x$i"`:
```
$ head /sys/bus/pci/devices/*/device
==> /sys/bus/pci/devices/0000:00:00.0/device <==
0x5904
...
```